stock_production_lot_warranty/models/stock_production_lot.py

```diff
@@ -15,34 +15,25 @@ class StockProductionLot(models.Model):
 
     warranty_exp_date = fields.Date(string="Warranty Expiration Date")
 
+    def _get_warranty_exp_date(self, start_date):
+        if not self.product_id:
+            return False
+        duration_type = self.product_id.product_tmpl_id.warranty_type
+        duration = self.product_id.product_tmpl_id.warranty
+        if not duration_type or not duration:
+            return False
+        if duration_type == "day":
+            warranty_delta = timedelta(days=duration)
+        elif duration_type == "week":
+            warranty_delta = timedelta(weeks=duration)
+        elif duration_type == "month":
+            warranty_delta = relativedelta(months=+duration)
+        elif duration_type == "year":
+            warranty_delta = relativedelta(years=+duration)
+        else:
+            raise RuntimeError(f"Invalid warranty_type {duration_type!r}")
+        return (start_date + warranty_delta).strftime(DEFAULT_SERVER_DATE_FORMAT)
```

Comment:
Is the `strftime` needed here? IIRC Odoo 15.0 allows to write Date/Datetime type without the need to convert to string.